src/hooks/useNamespace.js

```diff
@@ -0,0 +1,31 @@
+import { useModuleHierarchy } from '../components';
+
+// A hook which returns module namespace as a string
+// https://issues.folio.org/browse/STCOR-537
+
+// Example usage:
+
+// from app module (e.g. ui-users)
+// const namespace = useNamespace(); // "@folio/users"
+
+// from plugin embedded in app module (e.g. ui-plugin-find-order executing in ui-agreements)
+// const namespace = useNamespace(); // "@folio/agreements:@folio/plugin-find-order"
+
+// from plugin embedded in app module with `ignoreParents` option (e.g. plugin ui-plugin-find-order executing in ui-agreements)
+// const namespace = useNamespace({ ignoreParents: true }); // "@folio/plugin-find-order"
+
+// from plugin embedded in app module with `key` option present (e.g. ui-plugin-find-order executing in ui-agreements)
+// const namespace = useNamespace({ key: "filters-pane" }); // "@folio/agreements:@folio/plugin-find-order:filters-pane"
+const useNamespace = (options = {}) => {
+  const moduleHierarchy = useModuleHierarchy();
+  const { ignoreParents, key } = options;
+  let namespace = ignoreParents ? moduleHierarchy.pop() : moduleHierarchy.join(':');
+
+  if (key) {
+    namespace += `:${key}`;
```

Comment:
Maybe move `:` into `const NAMESPACE_DELIMITER = ':'`? That's either CSci 101 or a total jerk manager everything is fine but I have to put my thumbprint on it move. Your call :) I don't think it needs to be exported, though if `useGlobalNamespace` is implemented separately, maybe it would be helpful for it to be able to leverage that value? OTOH, if we're thinking of the string as opaque-but-user-friendly, then maybe it doesn't matter.

Reply:
Nice idea @zburke Thank you!